src/test/java/jp/kusumotolab/kgenprog/project/test/ParallelTestExecutorTest.java

```diff
@@ -0,0 +1,70 @@
+package jp.kusumotolab.kgenprog.project.test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import org.junit.Test;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import jp.kusumotolab.kgenprog.project.GeneratedSourceCode;
+
+public class ParallelTestExecutorTest {
+
+  @Test
+  public void testExec() {
+    final TestExecutor mockExecutor = mock(TestExecutor.class);
+    final AtomicBoolean called = new AtomicBoolean(false);
+    when(mockExecutor.exec(any())).then(e -> {
+      called.set(true);
+      return null;
+    });
+    final TestExecutor testExecutor = new ParallelTestExecutor(mockExecutor);
+    testExecutor.exec(null);
+    assertThat(called).isTrue();
```

Comment:
Mockitoに呼び出されたどうかを確認する機能があるのでそれを使うといいと思います。
（`Mockito.verify`、`Mockito.times` をstatic importしてください）
```suggestion
    verify(mockExecutor, times(1)).exec(any());
```